Added info time-delayed proxies (#4206)
w3f/polkadot-wiki: docs/learn/learn-proxies.md

import RPC from "./../../components/RPC-Connection"

:::caution The Account Tab in the Polkadot-JS UI cannot handle complicated proxy setups

> The Account Tab in the Polkadot-JS UI cannot handle complicated proxy setups (e.g. a proxy ->

Reviewer: ```suggestion
The Accounts Tab in the Polkadot-JS UI cannot handle complex proxy setups (e.g. a proxy ->
```

multisig -> an anonymous proxy which is part of another multisig). These complex setups must be done
using the [Extrinsics Tab](https://polkadot.js.org/apps/#/extrinsics) directly.

> These complex proxy setups should only be performed if you are comfortable enough interacting
> directly with the chain. **We recommend to use the Westend Testnet if you are testing features for
> the first time.**

Reviewer: ```suggestion
 **We recommend to use the [Westend Testnet](learn-DOT.md#getting-tokens-on-the-westend-testnet) if you are testing features for the first time.** By performing the complex proxy setups on the testnet, you can comfortably replicate the procedure on the main networks.
```

## Time-delayed Proxy

:::info

> See [this video tutorial](https://youtu.be/3L7Vu2SX0PE) to learn how you can use time-delayed

Reviewer: ```suggestion
See [this video tutorial](https://youtu.be/3L7Vu2SX0PE) to learn how you can setup and use time-delayed
```

We can add a layer of security to proxies by giving them a delay time. The delay will be quantified
in blocks. {{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} has approximately 6
seconds of block time. A delay value of 10 will mean ten blocks, which equals about one minute
delay. The proxy will announce its intended action using the `proxy.announce` extrinsic and will
wait for the number of blocks defined in the delay time before executing it. The proxy will include
the hash of the intended function call in the announcement. Within this time window, the intended
action may be canceled by accounts that control the proxy. This can be done by the proxy itself
using the `proxy.removeAnnouncement` extrinsic or by the proxied account using the the
`proxy.rejectAnnouncement` extrinsic. Now we can use proxies knowing that any malicious actions can
be noticed and reverted within a delay period. After the time-delay, the proxy can use the
`proxy.proxyAnnounced` extrinsic to execute the announced call.

:::info

> See [this video tutorial](https://youtu.be/LxIW7t15Iuw) to learn how you can setup and use

Reviewer: Provided feedback on the video on Element chat.

time-delayed proxies. The video goes through the example below.

:::

Let's take for example the stash account Eleanor that has a controller Charly. Eleanor does not
fully trust Charly, and as a consequence sets him as a time-delayed staking proxy. In this way, if
Charly submits an extrinsic to change the controller to Bob, such extrinsic can be rejected by
Eleanor. Bob can be even more malicious than Charly and change the reward destination to another
account he only controls. Remember that Eleanor added Charly as proxy, but she might not want to add
Bob as a controller. This implies that Eleanor monitors Charly, and that within the time-delay she

> can spot the announced extrinsic. As a monitor tool you can use a block explorer.

Reviewer: You need not use a block explorer. This information can be queried on-chain directly.

```suggestion
can spot the announced extrinsic. Eleanor can check all the proxy call announcements made on her account's behalf on-chain. On Polkadot-JS UI, go to Developer > Storage > Proxy > Announcements to view the list of unexecuted proxy calls on Eleanor's account.
```
